packages/unmock-core/src/service/service.ts

```diff
@@ -44,17 +46,21 @@ export class Service implements IService {
   constructor(opts: IServiceInput) {
     this.oasSchema = opts.schema;
     this.name = opts.name;
-    // Update the paths in the first level to regex if needed
-    if (this.schema === undefined || this.schema.paths === undefined) {
-      this.matcher = new OASMatcher({ schema: this.schema });
-      return; // empty schema or does not contain paths
-    }
-    this.updateSchemaPaths();
     this.hasPaths = // Find this once, as schema is immutable
       this.schema !== undefined &&
       this.schema.paths !== undefined &&
       Object.keys(this.schema.paths).length > 0;
-    this.matcher = new OASMatcher({ schema: this.schema });
+
+    if (this.hasDefinedPaths) {
+      this.buildSchemaRegExps();
```

Comment:
I like the pure `this.endpointToRegexp = Service.buildSchemaRegExps(this.schema)` more than this pattern of filling in stuff as side-effect, but I guess that's a matter of taste.

Reply:
That's fine, I'll update it as well.
Once I moved that it actually makes more sense to offload more of the matching to `OASMatcher`, so hold on for another re-PR please :)